src/core/metrics.cc

```diff
@@ -211,8 +211,8 @@ Metrics::InitializeNvmlMetrics()
               nvmlReturn_t nvmlerr =
                   nvmlDeviceGetPowerManagementLimit(gpu, &power_limit);
               if (nvmlerr != NVML_SUCCESS) {
-                LOG_ERROR << "failed to get power limit for GPU " << didx
-                          << ", NVML_ERROR " << nvmlerr;
+                LOG_WARNING << "failed to get power limit for GPU " << didx
```

Comment:
One other thing is that if there is one GPU that doesn't support the metrics, the warnings will be logged every time when the metrics is being updated. This will be annoying.

e.x.
```bash
E0204 21:50:51.341949 44 metrics.cc:213] failed to get power limit for GPU 0, NVML_ERROR 3
E0204 21:50:51.342005 44 metrics.cc:225] failed to get power usage for GPU 0, NVML_ERROR 3
E0204 21:50:51.342018 44 metrics.cc:238] failed to get energy consumption for GPU 0, NVML_ERROR 3
E0204 21:50:51.342028 44 metrics.cc:255] failed to get utilization for GPU 0, NVML_ERROR 3
E0204 21:50:53.345931 44 metrics.cc:213] failed to get power limit for GPU 0, NVML_ERROR 3
E0204 21:50:53.345993 44 metrics.cc:225] failed to get power usage for GPU 0, NVML_ERROR 3
E0204 21:50:53.346007 44 metrics.cc:238] failed to get energy consumption for GPU 0, NVML_ERROR 3
E0204 21:50:53.346020 44 metrics.cc:255] failed to get utilization for GPU 0, NVML_ERROR 3
...
```

Reply:
I updated metrics.cc to give up after a limited number of consecutive failures.